### PR TITLE
disable tag.gpgsign as it could be active for some systems

### PR DESCRIPTION
**Pull request by olamy**
Some test cases related to `tag` git command can fail in case of the following `~/.gitconfig` configuration
```
[tag]
       gpgSign = true
```

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/jenkinsci/git-plugin/blob/master/CONTRIBUTING.adoc) doc
- [x] I have referenced the Jira issue related to my changes in one or more commit messages
- [x] I have added tests that verify my changes
- [x] Unit tests pass locally with my changes
- [x] I have added documentation as necessary
- [x] No Javadoc warnings were introduced with my changes
- [x] No spotbugs warnings were introduced with my changes
- [x] Documentation in README has been updated as necessary
- [x] Online help has been added and reviewed for any new or modified fields
- [x] I have interactively tested my changes
- [x] Any dependent changes have been merged and published in upstream modules (like git-client-plugin)

## Types of changes

What types of changes does your code introduce? _Put an `x` in the boxes that apply. Delete the items in the list that do *not* apply_

- [x] Dependency or infrastructure update

### CODE REVIEW

**Review by rsandell — APPROVED** [review]

File: pom.xml
Comment (MarkEWaite) [reply pom.xml:105]: The git client plugin dependency on the master branch is now updated to [git client plugin 4.7.0](https://plugins.jenkins.io/git-client/releases/#version_4.7.0) that includes the test fixes for autosign.  Thanks again for making these improvements!

Comment (olamy) [reply pom.xml]: @MarkEWaite yeah. thanks for the ping ;) 
I update that here. But I still have one issue with with auto sign I need to figure out.
```
Expected: is <SUCCESS>
     but: was <FAILURE>
[ERROR] Errors: 
[ERROR]   CredentialsUserRemoteConfigTest.checkoutWithValidCredentials:273->Object.wait:328->Object.wait:-2 » TestTimedOut test timed out after 180 seconds

```

Comment (olamy) [reply pom.xml]: well this looks to be a flaky test

File: pom.xml
Comment (MarkEWaite) [inline pom.xml:105]: This can be removed since it is the version provided by the plugin bill of materials on the master branch.

File: pom.xml
Comment (MarkEWaite) [reply pom.xml]: I would be happy to approve the deletion of that test if it is flaky.  If I feel it is deeply important, I can rewrite it to not be flaky.  Let's not spend your time or mine on a flaky test that might block this pull request.

When I looked through other test runs, I didn't see other cases of that test failing, so I assume it was a one-time flake.  If in the future it fails more consistently, I'll delete it.

**Review by MarkEWaite — APPROVED** [review]
Thanks very much for your work on this change!

File: pom.xml
Comment (MarkEWaite) [reply pom.xml:105]: Removal suggestion
```suggestion
```

**Pull request merged**
